Linux Script/configure-linux.sh

```diff
@@ -582,32 +578,71 @@ elif [ "$RSYSLOG_VERSION_TMP" -ge "8" ]; then
-                                exit 1
+                                exit 1		    
 	    fi 
 	
     
-	    elif [ "$PKG_MGR" == "apt-get" ]; then
+	  elif [ "$PKG_MGR" == "apt-get" ]; then
 	
 				if [ $(dpkg-query -W -f='${Status}' rsyslog-gnutls 2>/dev/null | grep -c "ok installed") -eq 0 ]; then                            
-                                logMsgToConfigSysLog "ERROR" "ERROR: The rsyslog-gnutls package could not be installed automatically. Please install it and then run the script again. Manual instructions to configure rsyslog are available at https://www.loggly.com/docs/rsyslog-tls-configuration/. Rsyslog troubleshooting instructions are available at https://www.loggly.com/docs/troubleshooting-rsyslog/."
+                                logMsgToConfigSysLog "ERROR" "ERROR: The rsyslog-gnutls package could not be installed automatically. Please install it and then run the script again. Manual instructions to configure rsyslog are available at https://www.loggly.com/docs/rsyslog-tls-configuration/. Rsyslog troubleshooting instructions are available at https://www.loggly.com/docs/troubleshooting-rsyslog/."								
                                 exit 1
-                                fi		
-			
-	else
+                fi	
+      elif [ "$FORCE_SECURE" == "true" ]; then
+
+		  logMsgToConfigSysLog "WARN" "WARN: The rsyslog-gnutls package could not be download automatically because your package manager could not be found. Please install it and restart the rsyslog service to send logs to Loggly."
+	  else
+		     DEPENDENCIES_INSTALLED="false";
+	fi	
+    inputStr=$inputStrTls	
+fi
+}
 
-                      logMsgToConfigSysLog "WARN" "WARN: The rsyslog-gnutls package could not be download automatically because your package manager couldn't be found. Please download it manually for your distribution and then run the script again."					  
-    
-	fi				
-	inputStr=$inputStrTls
+#prompt users if they want to switch to insecure mode on gnutls-package download failure 
+switchToInsecureModeIfTLSNotFound()
+{
+if [ "$DEPENDENCIES_INSTALLED" == "false" ]; then
+
+    if [ "$SUPPRESS_PROMPT" == "false" ]; then
+
+            logMsgToConfigSysLog "WARN" "WARN: The rsyslog-gnutls package could not be download automatically because your package manager could not be found." 
+					  
+	        while true;
+			do
+	            read -p "Do you wish to continue with insecure mode? (yes/no)" yn
+						case $yn in
+						[Yy]* )
+							logMsgToConfigSysLog "INFO" "INFO: Going to overwrite the conf file: $LOGGLY_RSYSLOG_CONFFILE with insecure configuration";
```

Comment:
@mostlyjason This code will be executed whenever rsyslog-gnutls package couldn't be download or package manager is not found in the system. It will prompt the user to switch to insecure mode. For example, if I am in SUSE environment where the package manager is different from yum and apt-get so the rsyslog-gnutls package will not be downloaded. In that case, the script will prompt the user to switch to insecure mode.